libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/Media/AudioSender.cs

```diff
@@ -0,0 +1,177 @@
+            if (Track != null)
+            {
+                IsStreaming = false;
+                AudioStreamStopped.Invoke(this);
+
+                // Defer track destruction to derived classes.
+                DestroyLocalAudioTrack();
+                Debug.Assert(Track == null, "Implementation did not destroy the existing Track property yet did not throw any exception.", this);
+            }
+        }
+
+        /// <summary>
+        /// Internal callback invoked when the audio sender is attached to a transceiver created
+        /// just before the peer connection creates an SDP offer.
+        /// </summary>
+        /// <param name="audioTransceiver">The audio transceiver this sender is attached with.</param>
+        internal void AttachToTransceiver(Transceiver audioTransceiver)
+        {
+            Debug.Assert((Transceiver == null) || (Transceiver == audioTransceiver));
+            Transceiver = audioTransceiver;
+        }
+
+        /// <summary>
+        /// Internal callback invoked when a peer connection is about to create an offer,
+        /// and needs to create the audio transceivers and senders. The audio sender must
+        /// create a local audio track and attach it to the given transceiver.
+        /// </summary>
+        /// <returns>Once the asynchronous operation is completed, the <see cref="Track"/> property
+        /// must reference it.</returns>
+        internal async Task AttachTrackAsync()
+        {
+            Debug.Assert(Transceiver != null);
+
+            // Force again PreferredAudioCodec right before starting the local capture,
+            // so that modifications to the property done after OnPeerInitialized() are
+            // accounted for.
+            //< FIXME - Multi-track override!!!
+            Transceiver.PeerConnection.PreferredAudioCodec = PreferredAudioCodec;
+
+            // Ensure the local sender track exists
+            if (Track == null)
+            {
+                await CreateLocalTrackAsync();
+            }
+
+            // Attach the local track to the transceiver
+            if (Track != null)
+            {
+                Transceiver.LocalAudioTrack = Track;
+            }
+        }
+
+        internal void DetachTrack()
+        {
+            Debug.Assert(Transceiver != null);
+            Debug.Assert(Transceiver.LocalTrack == Track);
+            Transceiver.LocalAudioTrack = null;
+        }
+
+        /// <inheritdoc/>
+        protected override void MuteImpl(bool mute)
+        {
+            if (Track != null)
+            {
+                Track.Enabled = mute;
+            }
+        }
+
+        /// <summary>
+        /// Implement this callback to create the <see cref="Track"/> instance.
+        /// On failure, this method must throw an exception. Otherwise it must set the <see cref="Track"/>
+        /// property to a non-<c>null</c> instance.
+        /// </summary>
+        protected abstract Task CreateLocalAudioTrackAsync();
```

Comment:
Rename to `DoCreateLocalTrackAsync` or `CreateLocalTrackAsyncImpl`? So it's clear that it's supposed to be an implementation, the extra "Audio" doesn't say much.

Reply:
There's no `DoCreateLocalTrackAsync`, but do you think it needs the `Impl` anyway?

Reply:
Fixed.